rego/Chat.rego

```diff
@@ -241,39 +241,51 @@ if {
 #
 # Baseline GWS.CHAT.4.1v0.1
 #--
-GetFriendlyValue4_1(Value) := "OFF" if {
-    Value == "true"
-} else := "ON" if {
-    Value == "false"
-} else := Value
+default NoSuchEvent4_1(_) := true
+
+NoSuchEvent4_1(TopLevelOU) := true if {
+    Events := utils.FilterEvents(LogEvents, "RestrictChatProto restrictChatToOrganization", TopLevelOU)
+    count(Events) == 0
+}
+
+NoSuchEvent4_1(TopLevelOU) := true if {
+    # No such event...
+    Events := utils.FilterEvents(LogEvents, "RestrictChatProto externalChatRestriction", TopLevelOU)
+    count(Events) == 0
+}
 
 NonCompliantOUs4_1 contains {
     "Name": OU,
-    "Value": GetFriendlyValue4_1(LastEvent.NewValue)
+    "Value": "External chat is enabled for all domains"
 }
+
  if {
     some OU in utils.OUsWithEvents
-    Events := utils.FilterEvents(LogEvents,  "RestrictChatProto restrictChatToOrganization", OU)
-    # Ignore OUs without any events. We're already asserting that the
-    # top-level OU has at least one event; for all other OUs we assume
-    # they inherit from a parent OU if they have no events.
-    count(Events) > 0
-    LastEvent := utils.GetLastEvent(Events)
-    LastEvent.NewValue == "true"
+    Events_A := utils.FilterEvents(LogEvents, "RestrictChatProto restrictChatToOrganization", OU)
+    count(Events_A) > 0
+    LastEvent_A := utils.GetLastEvent(Events_A)
+    LastEvent_A.NewValue != "DELETE_APPLICATION_SETTING"
+
+    Events_B := utils.FilterEvents(LogEvents, "RestrictChatProto externalChatRestriction", OU)
+    count(Events_B) > 0
+    LastEvent_B := utils.GetLastEvent(Events_B)
+    LastEvent_B.NewValue != "DELETE_APPLICATION_SETTING"
+
+    LastEvent_A.NewValue != "true"
+    LastEvent_B.NewValue != "TRUSTED_DOMAINS"
 }
 
 tests contains {
     "PolicyId": "GWS.CHAT.4.1v0.1",
-    "Criticality": "Shall",
+    "Criticality": "Should",
```

Comment:
The criticality is staying SHALL, right? At least it's still a SHALL in the baseline md on this branch.
```suggestion
    "Criticality": "Shall",
```